doc/source/ray-core/examples/batch_training.ipynb

```diff
@@ -529,66 +573,111 @@
     "    except Exception:\n",
     "        return []\n",
     "\n",
-    "    loc_list = locdf[\"pickup_location_id\"].unique()\n",
+    "    pickup_location_ids = locdf[\"pickup_location_id\"].unique()\n",
     "\n",
-    "    group_refs = []\n",
-    "    for i in loc_list:\n",
+    "    data_batch_refs = []\n",
+    "    for pickup_location_id in pickup_location_ids:\n",
     "        # Put each data batch as a separate dataframe into Ray object store.\n",
-    "        group_refs.append(\n",
-    "            (i, ray.put(locdf.filter(pc.field(\"pickup_location_id\") == i).to_pandas()))\n",
+    "        data_batch_refs.append(\n",
+    "            (\n",
+    "                pickup_location_id,\n",
+    "                ray.put(\n",
```

Comment:
@stephanie-wang do you think we should use the experimental dynamic returns (https://docs.ray.io/en/master/ray-core/tasks/generators.html) in the public examples?

Reply:
Hmm yeah it'd be good to use it if we can. Or if the num_returns can be determined dynamically, we can just use the `return list` syntax.